### PR TITLE
fix: variable file fragments

### PR DESCRIPTION
**Pull request by wjones127**
Previous fix #2130 assumed that all fragments had same number of data files. This wasn't a great assumption. I've updated the fix and test accordingly.

### CODE REVIEW

File: rust/lance/src/io/commit.rs [hunk 1]
```diff
@@ -217,39 +217,39 @@ fn fix_schema(manifest: &mut Manifest) -> Result<()> {
...
+            .rev()
+            .flat_map(|file| file.fields.iter_mut())
+        {
+            if let Some(new_field_id) = old_field_id_mapping.get(field_id) {
+                if seen_fields.insert(*field_id) {
+                    *field_id = *new_field_id;
+                }
+            }
         }
     }
```
Comment (westonpace) [inline rust/lance/src/io/commit.rs:254]: Ok.  So "if a field id needs changed then replace the last instance of that field id in the fragment"?

**Review by westonpace — APPROVED** [review]
I'm glad we have you to work through these issues :laughing:  My head is spinning trying to work this migration out.

File: rust/lance/src/io/commit.rs (hunk 1 above)
Comment (wjones127) [reply rust/lance/src/io/commit.rs:254]: Yeah that's what I was thinking. I think it's technically possible for there to be more than 2 duplicate ids, but the last one should always be the one we want to keep.

Comment (westonpace) [reply rust/lance/src/io/commit.rs:254]: Wait...isn't the last one the one we are switching?

**Pull request merged**